scale-encode/Cargo.toml

```diff
@@ -14,22 +14,25 @@ keywords.workspace = true
 include.workspace = true
 
 [features]
-default = ["derive", "primitive-types", "bits"]
+default = ["std", "derive", "primitive-types", "bits"]
 
-# Include the derive proc macro
-derive = ["dep:scale-encode-derive"]
+# Activates std feature.
+std = ["scale-info/std"]
 
-# impls for key primitive-types
+# Include the derive proc macro. This is only available in std.
```

Comment:
I wonder what the main blockers are to the derive macro also being `no_std`; I can imagine annoyances around "is `extern crate alloc` imported or not" (I'd probably have a file that imports things from `alloc`/`core` or `std` depending on whether "std" feature is activated or not and use that)

Reply:
Haven't looked too deeply into that, I only checked some dependencies. I gathered that `syn` and `quote` are both std only crates. So I think it will require quite some workarounds.

Reply:
Nothing was blocking it on this side - that was my misinterpretation. Fixed that one.

Reply:
It doesn't matter if `syn` etc are not no-std anyways; they are only used at build time; as long as the generated code works fine on no_std it's all good (this assuming that you build on an std platform, which seems fair I think?) :)